### ISSUE TITLE
WebXR init accessing trackedAnchors causes error: "TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))"

### ISSUE THREAD

**Issue by benferns** [playcanvas/engine#5843]
This issue tracker is **only** for bug reports and feature requests. For general support/help, visit https://forum.playcanvas.com

For bug reports, include:

### Description

On trying to start WebXR on non-Android platforms, the error "TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))", or "undefined is not an object" occurs on the same line. Context below:

![Screenshot 2023-11-22 at 13 14 09](https://github.com/playcanvas/engine/assets/2927295/1d04034f-5a8b-4dd7-8602-ff72fda341d4)

So far I have confirmed this issue on my own iOS WebXR implementation ( https://launch.variant3d.com/docs ) and in WebXR emulator extensions. It appears not to happen on Android Chrome.

The change is recent (within the last few weeks) as I've just had a bunch of support requests pop-up this week around the issue.

### Steps to Reproduce

The fastest path to reproducing is to download a webxr emulator like https://chrome.google.com/webstore/detail/webxr-api-emulator/mjddjgeghkdijejnciaefnkjmkafnnje then follow the below:

1.  Create a blank WebXR starter project from the template
2. Publish the build with the latest version (1.66.3)
3. Hit the button to enter AR and observe the console errors
4. Delete the published build and change to previous minor version, or visit https://playcanv.as/p/XxDyZ9ZL/ (minor version pinned to 1.65.5)
5. Observe no errors in console

I can provide Variant Launch test projects/accounts if helpful, but since the same error shows across other implementations I think the underlying issue will be the same.

**Comment by Maksims** [comment]
Hi @benferns, thank you for the report. Do you have any reproduction using actual devices (not with emulators)? I have not been able to reproduce an issue. Based on the specs `trackedAnchors` should exist on XRFrame if platform supports XRAnchor: https://immersive-web.github.io/anchors/

I have tried using link to the project you've provided on Windows Chrome, but the extension you've provided does not seem to implement WebXR AR polyfill.

Can you verify there are no other extensions running in your browser that might implement the WebXR polyfill?

**Comment by benferns** [comment]
Hi Maksims - just to update this, I've resolved the issue on my end by making anchors non opt-in for users of my SDK (previously it needed to be explicitly enabled). I might be able to shed some more light on the underlying cause though:

Here's the console logs from the immersive web anchors example - you can see anchors are requested, and the session denied as they aren't supported.
<img width="1132" alt="image" src="https://github.com/playcanvas/engine/assets/2927295/c4fd7443-fc37-4ecd-9b69-32c150409aa2">

On the playcanvas repro logs, light estimation is denied (as an optional feature), but anchors is not mentioned as required, or optional, which might be the issue:
<img width="1146" alt="image" src="https://github.com/playcanvas/engine/assets/2927295/9c3c9ea8-1c82-4abe-862e-2ed8569ca4dd">

I suspect the detection of anchor support is somehow set as true if a polyfill is used, even if the actual requestSession doesn't allow it. This could maybe be clarified by adding 'anchors' to the requiredFeatures list so the error is thrown at session creation, or the internal support flag updated on requestSession to be false.

**Comment by Maksims** [comment]
Hi @benferns, I've investigated more into that, and can confirm that when `anchors` were not requested for the session, it would still try to iterate through anchors properties on XRFrame. But only if anchors API is supported by the platform.
As I follow the implementation of [Specs](https://immersive-web.github.io/anchors/) very precisely, it worked well, as `tackedAnchors` property should be defined on XRFrame, regardless if `anchors` feature is requested for the session, as far as anchors API is supported generally - [this is what Specs states](https://immersive-web.github.io/anchors/#dom-xrframe-trackedanchors) (notice that this property is not optional).

**In short: polyfill implementation - is the one that leads to issues.**

Please ensure your implementation also will follow specs, which basically is pretty much true for all WebXR extensions I've read: if API is supported then properties should be defined on their objects (accessors on prototypes) and in type specified in specs, regardless if feature is requested for the session or not.

Nevertheless I've implemented a PR that will take such case into the account, and will prevent attempt to access anchors API in case when it is not enabled on the session: https://github.com/playcanvas/engine/pull/5879
